client/src/components/Connections/ListConnections.tsx

```diff
@@ -4,72 +4,71 @@ import { ConnectionInterface } from '../../App'
 const ListConnections = () => {
   const [connections, setConnections] = useState<any | any[]>([]);
 
-  const getConnections = async () => {
-    try {
-      const response = await fetch("http://localhost:3000/connections");
-      const jsonData = await response.json();
+  const getConnections = () => {
 
-      const tuples: any[] = [];
-      await jsonData.rows.map(async (connection:ConnectionInterface) => {
-        const u1 = await getUserNamePromise(connection.user1_id);
-        const u2 = await getUserNamePromise(connection.user2_id);
-        tuples.push({ user1_id: connection.user1_id, user1_name: u1, user2_name: u2, user2_id: connection.user2_id });
-        setConnections(tuples);
-      });
+    const response = fetch("http://localhost:3000/connections");
+    response.then((res) => {
+      const jsonData = res.json();
+      jsonData.then((resJson) => {
+        const promisesArray = resJson.rows.map((connection: ConnectionInterface) => {
+          const user1_id = Promise.resolve(connection.user1_id);
+          const name1Promise = getUserNamePromise(connection.user1_id);
```

Comment:
You can use a then here to return a promise with the structure that you need, instead of using the `Promise.Resolve` it feels a bit hacky.

```suggestion
          const user1_id = Promise.resolve(connection.user1_id);
          const name1Promise = getUserNamePromise(connection.user1_id).then((userName) => {user1_id:connection.user1_id, user1_name:userName});
```
Also, will be good to consider using types for these structs. https://www.typescriptlang.org/docs/handbook/interfaces.html